R/clean_data_helpers.R

```diff
@@ -66,11 +77,22 @@ scan_columns <- function(x) {
 #'                              package = "cleanepi"))
 #' )
 scan_data <- function(data) {
-  scan_result           <- data.frame(t(apply(data, 2L, scan_columns)))
-  names(scan_result)    <- c("missing", "numeric", "date", "character",
+  # when scanning through the data, logical and factor columns will be treated
+  # differently from the others. It means only the percent of missing and Date
+  # values will be evaluated for these columns.
+  # The percent of numeric and character value will be set automatically to 0 as
+  # to prevent from the effects of the conversion to numeric and character.
+  #
+  types         <- as.character(vapply(data, class, character(1L)))
+  scan_result   <- NULL
+  j             <- 1
+  for (i in names(data)) {
+    scan_result <- rbind(scan_result, scan_columns(data[[i]], types[j]))
+    j           <- j + 1
+  }
```

Comment:
data.frames, matrices & vectors should never be grown in R.

In other words, `for` loops should never rely on iteratively using `c()` or `rbind()`.

```suggestion
  scan_result <- vapply(seq_len(ncol(data)), function(col_index) {
    scan_columns(data[[col_index]], types[[col_index]])
   }, numeric(5))
```

Reply:
I have account for this in commit `828534d`.